src/sender.rs

```diff
@@ -88,8 +100,12 @@ impl Sender {
         }
     }
 
-    pub fn set_encryption_key(&mut self, key_material: &[u8]) -> Result<()> {
-        self.secret = Some(KeyMaterial(key_material).expand_as_secret(&self.cipher_suite)?);
+    pub fn set_encryption_key<M>(&mut self, key_material: &M) -> Result<()>
+    where
+        M: AsRef<[u8]> + ?Sized,
```

Comment:
`K` for key?

Reply:
actually, M is fine, you wanna distinguish the "id" from the "material" right?

Reply:
I've named the parameters more specificly